pkg/controller/gcp/compute/gke.go

```diff
@@ -143,12 +144,22 @@ func (r *Reconciler) _connect(instance *gcpcomputev1alpha1.GKECluster) (gke.Clie
 	if err != nil {
 		return nil, err
 	}
-
-	creds, err := gcp.ProviderCredentials(r.kubeclient, p, gke.DefaultScope)
+	secret := &corev1.Secret{}
+	name := meta.NamespacedNameOf(&corev1.ObjectReference{
+		Name:      p.Spec.Secret.Name,
+		Namespace: p.Namespace,
+	})
+	if err := r.Client.Get(context.TODO(), name, secret); err != nil {
```

Comment:
This will be remedied when all reconcilers are moved to the shared managed reconciler (which passes `ctx`) and should not be considered blocking, but it would be nice to use the same `context` (either `TODO` or `Background`) consistently throughout the controller and maybe make it a `var ctx` at beginning of file. Feel free to leave as is for now though.